drools-wb-screens/drools-wb-test-scenario-editor/drools-wb-test-scenario-editor-backend/src/main/java/org/drools/workbench/screens/testscenario/backend/server/ScenarioRunnerService.java

```diff
@@ -110,22 +111,30 @@ public TestScenarioResult run(final String identifier,
     public List<TestResultMessage> runAllTests(final String identifier,
                                                final Path path) {
         try {
-            final List<Scenario> scenarios = scenarioLoader.loadScenarios(path);
-
-            ScenarioRunner4JUnit scenarioRunner = new ScenarioRunner4JUnit(
-                    scenarios,
-                    getKSessions(path,
-                                 scenarios),
-                    getMaxRuleFirings());
+            ArrayList<TestResultMessage> result = new ArrayList<>();
+            final Map<Path, Scenario> scenarios = scenarioLoader.loadScenarios(path);
+
+            Map<String, KieSession> kSessions = getKSessions(path,
+                                                             scenarios.values());
+
+            for (Map.Entry<Path, Scenario> entry : scenarios.entrySet()) {
+                ScenarioRunner4JUnit scenarioRunner = new ScenarioRunner4JUnit(
+                        entry.getValue(),
+                        kSessions,
+                        getMaxRuleFirings());
+                result.add(run(identifier,
+                               entry.getKey(),
+                               scenarioRunner));
+            }
```

Comment:
Minor comment, just for consistency, could we use `final` for all vars?
```java
final ArrayList<TestResultMessage> result = new ArrayList<>();
            final Map<Path, Scenario> scenarios = scenarioLoader.loadScenarios(path);

            final Map<String, KieSession> kSessions = getKSessions(path,
                                                                   scenarios.values());

            for (Map.Entry<Path, Scenario> entry : scenarios.entrySet()) {
                final ScenarioRunner4JUnit scenarioRunner = new ScenarioRunner4JUnit(
                        entry.getValue(),
                        kSessions,
                        getMaxRuleFirings());
                result.add(run(identifier,
                               entry.getKey(),
                               scenarioRunner));
            }
```

Reply:
We should, updating.

Reply:
Updated.